Player/Player.gd

```diff
@@ -2,26 +2,35 @@ extends CharacterBody2D
 
 signal interacted_with(interactable : Interactable)
 
-const SPEED = 50.0
+const ACCELERATION = 800
+const SPEED = 50
+const FRICTION = 300
+
 var interactables = []
 var closest_interactable
 
+@onready var animationPlayer = $AnimationPlayer
+@onready var animationTree = $AnimationTree
+@onready var animationState = animationTree.get("parameters/playback")
+
 func _physics_process(delta):
+    
     # Get the input direction and handle the movement/deceleration.
     # As good practice, you should replace UI actions with custom gameplay actions.
     var lr_direction = Input.get_axis("ui_left", "ui_right")
-
-    if lr_direction:
-        velocity.x = lr_direction * SPEED
-    else:
-        velocity.x = move_toward(velocity.x, 0, SPEED)
-
     var ud_direction = Input.get_axis("ui_up", "ui_down")
+    var input_vector = Vector2(lr_direction, ud_direction)
+    input_vector = input_vector.normalized()
 
-    if ud_direction:
-        velocity.y = ud_direction * SPEED
+    if input_vector != Vector2.ZERO:    
+        animationTree.set("parameters/Walk/blend_position", input_vector)       
+        animationTree.set("parameters/Idle/blend_position", input_vector)
```

Comment:
This tells our AnimationTree what parameter is being used to define the "blend space" coordinate. The input_vector (the direction the player is indicating they want to move toward) is mapped into the blend space, which we then play whatever animation is closest to our input vector in the space. It'll make more sense if you watch this video:
https://www.youtube.com/watch?v=Z9aR9IiiHT8&list=PL9FzW-m48fn2SlrW0KoLT4n5egNdX-W9a&index=6